op-e2e/e2eutils/fakebeacon/blobs.go

```diff
@@ -0,0 +1,194 @@
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// FakeBeacon presents a beacon-node in testing, without leading any chain-building.
+// This merely serves a fake beacon API, and holds on to blocks,
+// to complement the actual block-building to happen in testing (e.g. through the fake consensus geth module).
+type FakeBeacon struct {
+	log log.Logger
+
+	// directory to store blob contents in after the blobs are persisted in a block
+	blobsDir  string
+	blobsLock sync.Mutex
+
+	beaconSrv         *http.Server
+	beaconAPIListener net.Listener
+
+	genesisTime uint64
+	blockTime   uint64
+}
+
+func NewBeacon(log log.Logger, blobsDir string, genesisTime uint64, blockTime uint64) *FakeBeacon {
+	return &FakeBeacon{
+		log:         log,
+		blobsDir:    blobsDir,
+		genesisTime: genesisTime,
+		blockTime:   blockTime,
+	}
+}
+
+func (f *FakeBeacon) Start(addr string) error {
+	listener, err := net.Listen("tcp", addr)
+	if err != nil {
+		return fmt.Errorf("failed to open tcp listener for http beacon api server: %w", err)
+	}
+	f.beaconAPIListener = listener
+
+	mux := new(http.ServeMux)
+	mux.HandleFunc("/eth/v1/beacon/genesis", func(w http.ResponseWriter, r *http.Request) {
+		err := json.NewEncoder(w).Encode(&eth.APIGenesisResponse{Data: eth.ReducedGenesisData{GenesisTime: eth.Uint64String(f.genesisTime)}})
+		if err != nil {
+			f.log.Error("genesis handler err", "err", err)
+		}
+	})
+	mux.HandleFunc("/eth/v1/config/spec", func(w http.ResponseWriter, r *http.Request) {
+		err := json.NewEncoder(w).Encode(&eth.APIConfigResponse{Data: eth.ReducedConfigData{SecondsPerSlot: eth.Uint64String(f.blockTime)}})
+		if err != nil {
+			f.log.Error("config handler err", "err", err)
+		}
+	})
+	mux.HandleFunc("/eth/v1/beacon/blob_sidecars/", func(w http.ResponseWriter, r *http.Request) {
+		blockID := strings.TrimPrefix(r.URL.Path, "/eth/v1/beacon/blob_sidecars/")
+		slot, err := strconv.ParseUint(blockID, 10, 64)
+		if err != nil {
+			f.log.Error("could not parse block id from request", "url", r.URL.Path)
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+		bundle, err := f.LoadBlobsBundle(slot)
+		if err != nil {
+			f.log.Error("failed to load blobs bundle", "slot", slot)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		query := r.URL.Query()
+		rawIndices := query["indices"]
+		indices := make([]int, 0, len(bundle.Blobs))
+		if len(rawIndices) == 0 {
+			// request is for all blobs
+			for i := range bundle.Blobs {
+				indices = append(indices, i)
+			}
+		} else {
+			for _, raw := range rawIndices {
+				ix, err := strconv.ParseUint(raw, 10, 64)
+				if err != nil {
+					f.log.Error("could not parse index from request", "url", r.URL)
+					w.WriteHeader(http.StatusBadRequest)
+					return
+				}
+				indices = append(indices, int(ix))
+			}
+		}
+
+		var mockBeaconBlockRoot [32]byte
+		mockBeaconBlockRoot[0] = 42
+		binary.LittleEndian.PutUint64(mockBeaconBlockRoot[32-8:], slot)
```

Comment:
We probably should have a single shared function for creating the mock beacon block root. It feels pretty weird that its different between here and in fakepos.